### PR TITLE
Fix: Fixed issue with creating shortcuts in libraries

### LINKED ISSUE #10726: Bug: Creating shortcuts in a library directory does not work
### Description

As the title suggests, when in a library folder such as Documents, Music, Pictures, or Videos, shortcuts cannot be created of files nor folders.

The error message from the debug.log when trying to do so states:
```
2022-12-15 13:18:15.5286|WARN|CreateOrUpdateLinkAsync|The system cannot find the path specified. (0x80070003)
	System.IO.DirectoryNotFoundException: The system cannot find the path specified. (0x80070003)
   at System.Runtime.InteropServices.ComTypes.IPersistFile.Save(String pszFileName, Boolean fRemember)
   at Vanara.Windows.Shell.ShellLink.SaveAs(String linkFile)
   at Files.App.Helpers.FileOperationsHelpers.CreateOrUpdateLinkAsync(String linkSavePath, String targetPath, String arguments, String workingDirectory, Boolean runAsAdmin)
```

### Steps To Reproduce

1. Navigate to a library folder via the left side menu.
2. Create a shortcut of the desired file or folder.
3. Nothing happens

### Steps to fix

I believe this may be due to the difference in the displayed path when in a library folder. That is, instead of the regular path one might expect, the path is:
```
C:\Users\Username\AppData\Roaming\Microsoft\Windows\Libraries\Documents.library-ms
```
My guess is that this pathname difference is in conflict with the current way of creating shortcuts and it cannot resolve the path to create the desired shortcut. Correctly resolving the path might fix the issue.

### Files Version

Version: 2.4.1.0 OS Version: 10.0.19044.2364

### Windows Version

Windows 10 21H2 19044.2364

### Relevant Assets/Logs

[debug.log](https://github.com/files-community/Files/files/10240422/debug.log)

### PR DESCRIPTION
**Pull request by hecksmosis**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #10726 

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [ ] Tested the changes for accessibility

### CODE REVIEW

**Review by ferrariofilippo — COMMENTED** [review]
LGTM and it works!
This fix should be added also when `CreateShortcutDialogViewModel` is instantiated (added in #10879)

File: src/Files.App/Interacts/BaseLayoutCommandImplementationModel.cs [hunk 1]
```diff
@@ -84,9 +84,16 @@ public virtual void RenameItem(RoutedEventArgs e)
...
+				!library.IsEmpty)
+			{
+				currentPath = library.DefaultSaveFolder;
+			}
+
 			foreach (ListedItem selectedItem in SlimContentPage.SelectedItems)
 			{
-				var filePath = Path.Combine(associatedInstance.FilesystemViewModel.WorkingDirectory,
+				var filePath = Path.Combine(currentPath,
 								string.Format("ShortcutCreateNewSuffix".GetLocalizedResource(), selectedItem.Name) + ".lnk");
```
Comment (QuaintMako) [inline src/Files.App/Interacts/BaseLayoutCommandImplementationModel.cs:97]: The file name could be composed in another variable, and then combine in `filePath` to gain a bit of readability.

Comment (hecksmosis) [reply src/Files.App/Interacts/BaseLayoutCommandImplementationModel.cs]: Done

**Review by QuaintMako — DISMISSED** [review]

**Comment by ferrariofilippo** [comment]
> This fix should be added also when `CreateShortcutDialogViewModel` is instantiated (added in #10879)

Can you fix this? Otherwise we would need a different PR to fix the same bug

**Comment by hecksmosis** [comment]
Done.

**Review by ferrariofilippo — APPROVED** [review]
LGTM

**Review by QuaintMako — APPROVED** [review]

**Pull request merged**